### PR TITLE
Allow printing overridden config values via `--include-overridden`.

### LINKED ISSUE #2828: Support  `--include-overridden` argument to `config list` command.
Print overrides values with "commented out" like `# ui.editor="editor from user config"`

### PR DESCRIPTION
**Pull request by daehyeok**
fix #2828 

# Example

```shell
 ~/workspace/jj ❯ ./target/debug/jj config list       
ui.editor="emacs -nw -q"
ui.pager="bat -l toml --plain"
user.name="Daehyeok Mun"
git.fetch="upstream"
git.push="orgin"
user.email="daehyeok@google.com"

 ~/workspace/jj ❯ ./target/debug/jj config list  --include-overridden
# ui.editor="nano"
# ui.pager="less -R"
ui.editor="emacs -nw -q"
ui.pager="bat -l toml --plain"
# user.email="daehyeok@gmail.com"
user.name="Daehyeok Mun"
git.fetch="upstream"
git.push="orgin"
user.email="daehyeok@google.com"
```
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/commands/config.rs [hunk 1]
```diff
@@ -150,18 +153,23 @@ pub(crate) fn cmd_config_list(
         is_overridden,
     } in &values
     {
-        // Remove overridden values.
-        // TODO(#1047): Allow printing overridden values via `--include-overridden`.
+        let mut comment_out = false;
```
Comment (martinvonz) [inline cli/src/commands/config.rs:156]: It looks like this value always equals `*is_overridden`, so can we just use that on line 172 instead? I think it still reads well. Or do will there be other reasons to comment out a line in the future?

**Review by martinvonz — APPROVED** [review]

File: cli/src/commands/config.rs [hunk 2]
```diff
@@ -78,6 +78,9 @@ pub(crate) struct ConfigListArgs {
     /// Whether to explicitly include built-in default values in the list.
     #[arg(long)]
     pub include_defaults: bool,
+    /// Allow printing overridden values.
```
Comment (martinvonz) [inline cli/src/commands/config.rs:81]: Oh, could you also mention this new feature in the changelog?

File: cli/src/commands/config.rs (hunk 1 above)
Comment (daehyeok) [reply cli/src/commands/config.rs]: you right. i don't know why i though i need additional variable ...

File: CHANGELOG.md [hunk 3]
```diff
@@ -22,6 +22,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * There's now a virtual root operation, similar to the [virtual root
   commit](docs/glossary.md#root-commit). It appears at the end of `jj op log`.
 
+* `jj config list` gained a `--include-overridden` option to allow
+  printing overridden config values
```
Comment (martinvonz) [inline CHANGELOG.md:26]: nit: missing period (we seem to quite consistently have period at end of sentence here)

**Pull request merged**